### DISCUSSION TITLE
Simpler instructions for setting up remote repo after using "upload from file"

### DISCUSSION THREAD

**Discussion by Kentoseth** [WeblateOrg/weblate#8080] [Q&A]
I investigated here:

https://docs.weblate.org/en/weblate-4.14/vcs.html#vcs-repos

and tried to look at all the documentation linked from:

https://url/weblate/settings/project/repo/#vcs

But to configure a remote repo is still vague and confusing.

I created my repo by using: "upload from file" and set it to local (no remote).

Now I want to store the translated data as:

 - A public repo
 - On Codeberg (or does multi-push work?)
 - That pushes the updated translations infrequently (not per save, but if this is easier, than that is fine too) or manually

1. Do I need to setup the remote repo first?
2. Do I generate an SSH keypair locally or via weblate? Do I need an SSH keypair for a public repo?
3. What is the difference between 'git' and "git with force push"?
4. Is there a simpler how-to guide to set it up? The documentation explains what is going on, but there is very little 'how'.

**Comment by Kentoseth** [comment]
I am still unable to figure this out after spending some time trying to.

This is the attempted workflow I tried (which is sadly all over the place, as the documentation keeps linking to other documentation with no clear flow):

1. Create new weblate component via file upload (.ODT file), example: es.odt (Spanish)
2. Add English translation (en.odt) with some translations to it to create: es.odt and en.odt
3. Go to: https://localhost/weblate/manage/ssh/
    - Add host key: codeberg.org
    - Copy public key FROM weblate TO codeberg
4. Create a repo on remote code-hosting site (eg. codeberg)
5. The repo was not working when blank, so initialize it with a README.md (either locally on the PC or via the code-hosting site)
6. With the new repo available remotely, go to: https://localhost/weblate/settings/my-project/weblate-test/#vcs
7. - VCS: git
    - Source code repo: git@codeberg.org:user/translation_test.git
    - branch: master
    - repo push URL: git@codeberg.org:user/translation_test.git
    - Leave everything else blank

Even after somehow managing to figure this out, I get an error in the `Files` tab after this saying that the `es.odt` file is not found. So it didn't even work.

Is it just not possible to create a remote repo from an existing local weblate repo?

I have existing translations that I wish to backup remotely and the process of doing so is more complicated than it needs to be.

**Reply by nijel** [reply]
Weblate creates repo with `main` as default branch. Once you have changed branch to `master`, Weblate fetched the remote repository and switched to a different branch, which contained no documents.

The safest approach is:

1. Create a blank repo on code-hosting site
2. Clone repo from Weblate locally (for example `git clone https://hosted.weblate.org/git/weblate/application/`)
3. Push the Weblate repo to the code-hosting site
4. Change Weblate configuration to the code-hosting site

This way you will have one history in the repository.
